src/main/java/jewellery/inventory/service/UserInOrganizationService.java

```diff
@@ -0,0 +1,191 @@
+package jewellery.inventory.service;
+
+import java.util.List;
+import java.util.UUID;
+import jewellery.inventory.aspect.EntityFetcher;
+import jewellery.inventory.aspect.annotation.LogCreateEvent;
+import jewellery.inventory.aspect.annotation.LogDeleteEvent;
+import jewellery.inventory.aspect.annotation.LogUpdateEvent;
+import jewellery.inventory.dto.request.UserInOrganizationRequestDto;
+import jewellery.inventory.dto.response.OrganizationMembersResponseDto;
+import jewellery.inventory.dto.response.OrganizationSingleMemberResponseDto;
+import jewellery.inventory.exception.organization.MissingOrganizationPermissionException;
+import jewellery.inventory.exception.organization.UserIsNotPartOfOrganizationException;
+import jewellery.inventory.mapper.OrganizationMapper;
+import jewellery.inventory.model.*;
+import jewellery.inventory.repository.OrganizationRepository;
+import jewellery.inventory.repository.UserInOrganizationRepository;
+import jewellery.inventory.service.security.AuthService;
+import lombok.AllArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserInOrganizationService implements EntityFetcher {
+  private static final Logger logger = LogManager.getLogger(UserInOrganizationService.class);
+  private final OrganizationRepository organizationRepository;
+  private final OrganizationMapper organizationMapper;
+  private final AuthService authService;
+  private final UserService userService;
+  private final UserInOrganizationRepository userInOrganizationRepository;
+  private final OrganizationService organizationService;
+
+  public OrganizationMembersResponseDto getAllUsersInOrganization(UUID organizationId) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    validateUserInOrganization(organization);
+    return organizationMapper.toOrganizationMembersResponseDto(organization);
+  }
+
+  @LogUpdateEvent(eventType = EventType.ORGANIZATION_USER_UPDATE)
+  public OrganizationSingleMemberResponseDto updateUserPermissionsInOrganization(
+      UUID userId, UUID organizationId, List<OrganizationPermission> organizationPermissionList) {
+
+    UserInOrganization userInOrganization =
+        userInOrganizationRepository
+            .findByUserIdAndOrganizationId(userId, organizationId)
+            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));
+
+    Organization organization = userInOrganization.getOrganization();
+    User userForUpdate = userInOrganization.getUser();
+
+    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
```

Comment:
we should do the validation before fetching data from the db.
```suggestion
    validateCurrentUserPermission(organization, OrganizationPermission.MANAGE_USERS);
    UserInOrganization userInOrganization =
        userInOrganizationRepository
            .findByUserIdAndOrganizationId(userId, organizationId)
            .orElseThrow(() -> new UserIsNotPartOfOrganizationException(userId, organizationId));

    Organization organization = userInOrganization.getOrganization();
    User userForUpdate = userInOrganization.getUser();
```